File: VB/CustomClustererSample/Tree.vb
Imports System

Namespace CustomClustererSample
    Friend Class Tree

        Private locationName_Renamed As String

        Public Sub New()
            locationName_Renamed = ""
        End Sub
        Public Property Latitude() As Double
        Public Property Longitude() As Double
        Public Property LocationName() As String
            Get
                Return Me.locationName_Renamed
            End Get
            Set(ByVal value As String)
                If value Is Nothing Then
                    Throw New ArgumentNullException("LocationName")
                End If
                If value.Equals(Me.locationName_Renamed) Then
                    Return
                End If
                Me.locationName_Renamed = value
            End Set
        End Property
    End Class
End Namespace

File: CS/CustomClustererSample/Tree.cs
using System;

namespace CustomClustererSample {
    class Tree {
        string locationName;

        public Tree() {
            locationName = "";
        }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string LocationName {
            get { return this.locationName; }
            set {
                if (value == null) throw new ArgumentNullException("LocationName");
                if (value.Equals(this.locationName)) return;
                this.locationName = value;
            }
        }
    }
}
